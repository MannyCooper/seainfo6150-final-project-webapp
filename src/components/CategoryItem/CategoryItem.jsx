import React from "react";
import { Link } from "react-router-dom";
import styles from "./CategoryItem.module.css";

const CategoryItem = (props) =>{
    let path = '/categories/' + props.category[0];
    return (
        <div className={styles.category}>
            <Link className={styles.link} to={path} >
                <span className={styles.name}>{props.category[0]}</span>
                <div className={styles.number_wrap}>
                    <span className={styles.number}>{props.category[1]}</span>
                </div>
            </Link>
        </div>
    )
}

export default CategoryItem;