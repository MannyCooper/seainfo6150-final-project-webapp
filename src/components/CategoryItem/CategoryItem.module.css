.category{
    margin-right: 20px;
    transition: background-color .3s,transform .4s cubic-bezier(.4,0,.25,1);
    border-radius: 25px;
    background-color: #f2f2f7;
}

.category:hover{
    -webkit-transform: translateY(-6px);
    transform: translateY(-6px);
    -webkit-transform: scale(1.3);
    transform: scale(1.1);
    background-color: #ff375f;
}
.category:hover .name{
    color: white;
}

.link{
    display: flex;
    justify-content: space-between;
}

.name{
    color: #3a3a3c;
    line-height: 4;
    padding: 10px 15px;
}

.number{
    background: #e5e5ea;
    color: #48484a;
    border-radius: 65px;
    width: 40px;
    height: 40px;
    text-align: center;
    line-height: 40px;
    margin-right: 15px;
}

.number_wrap{
    display: flex;
    align-items: center;
}

@media only screen and (max-width: 768px){
    .category{
        margin-right: 6px;
    }
    .name{
        padding: 5px 10px
    }
}
