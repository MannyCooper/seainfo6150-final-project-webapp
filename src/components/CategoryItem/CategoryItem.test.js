 /* global render */
 import React from "react";
 import CategoryItem from "./CategoryItem.jsx";
 
 const testCategory = ['Test',10]
 
 it("renders the CategoryItem component correctly", () => {
   const { container } = render(<CategoryItem category={testCategory} />);
   expect(container).toMatchSnapshot();
 });
 