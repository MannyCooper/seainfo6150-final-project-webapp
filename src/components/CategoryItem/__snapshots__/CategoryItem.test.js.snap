// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CategoryItem component correctly 1`] = `
<div>
  <div
    class="category"
  >
    <a
      class="link"
      href="/categories/Test"
    >
      <span
        class="name"
      >
        Test
      </span>
      <div
        class="number_wrap"
      >
        <span
          class="number"
        >
          10
        </span>
      </div>
    </a>
  </div>
</div>
`;
